airbyte-cdk/python/airbyte_cdk/sources/declarative/schema/json_schema.py

```diff
@@ -33,8 +34,38 @@ def __post_init__(self, options: Mapping[str, Any]):
 
     def get_json_schema(self) -> Mapping[str, Any]:
         json_schema_path = self._get_json_filepath()
-        with open(json_schema_path, "r") as f:
-            return json.loads(f.read())
+        resource, schema_path = self.extract_resource_and_schema_path(json_schema_path)
```

Comment:
I don't love this solution of splitting the URL. As mentioned in the below comment, this is a slight hack for us to figure out what resource the schemas are mounted under. Because setup.py mounts them in the `source_sentry` resource for example, we need to figure out from the `airbyte_cdk` module what the original source name was. All existing configs already prefix the source name so we can extract it from the first part of the path.

An alternative that I thought about was adding a new field to the JsonSchema dataclass called resource. However, I didn't like that because it will in 100% of cases always end up being the same as the name of the connector. It's also a weak abstraction and exposes our underlying file parsing mechanism by breaking it into two parts.